docs/matrix.md

```diff
@@ -0,0 +1,20 @@
+<!--
```

Comment:
copyright license please 🙏

Reply:
🤔 there are no copyright licenses on any other docs in https://github.com/tektoncd/pipeline/tree/main/docs (spot checked)

Reply:
Thanks @jerop for pointing that out, I have opened a github discussion for the same:

https://github.com/tektoncd/pipeline/discussions/4626